test/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs

```diff
@@ -29,6 +29,8 @@ public class TestProject
 
         public List<string> References { get; } = new List<string>();
 
+        public PackageReference PublishedNuGetPackageLibrary { get; set; }
```

Comment:
I think we should represent the package references differently.  Instead of putting the referenced packages in the `ReferencedProjects` list, create a new `PackageReferences` property on `TestProject`.  Then you can remove the `PublishedNuGetPackageLibrary` property here and simply iterate over each list separately when processing the references.